### PR TITLE
Coldstaking pool fee

### PR DESCRIPTION
**Pull request by aguycalled**
This PR introduces a consensus-based activation of the ability for staking pools to take a fee out of the staking reward to an arbitrary address.

At the same time, it fixes a bug which would allow a staking pool to contribute the whole inputs' amount to the Community Fund.

Version Bit: 18
Starting Voting: Jun 1st, 2019
End Voting: Jun 1st, 2021

A pool operator can use the node arguments `-poolfee` (range: 0-100) and `-pooladdress` to define the percentage taken out of the reward and the address where the fee is accumulated.

### CODE REVIEW

**Comment by aguycalled** [comment]
Pending on functional tests pr for adding a test.

**Comment by proletesseract** [comment]
utACK

File: src/versionbits.h [hunk 1]
```diff
@@ -33,7 +33,8 @@ static const int32_t VERSIONBITS_NUM_BITS = 29;
 * Bit 7 -> C FUND ACCUMULATION
 * Bit 8 -> NTP SYNC
 * Bit 14 -> C FUND ACCUMULATION SPREAD
-* Bit 17 -< C FUND REDUCED QUORUM
+* Bit 17 -> C FUND REDUCED QUORUM
+* Bit 17 -> COLD STAKING POOL FEE
```
Comment (mxaddict) [inline src/versionbits.h]: Is this supposed to be BIT 17?

Comment (aguycalled) [reply src/versionbits.h]: Yes, it should be Bit 18. It's a typo

**Comment by mxaddict** [comment]
utACK

**Comment by mxaddict** [comment]
I fixed the `push_back(Pair)` (Replaced with `pushKV()`)

**Comment by mxaddict** [comment]
untestedACK

**Comment by mxaddict** [comment]
Was able to build locally (Ubuntu 19.10 dev branch)

Running test suite now.

**Comment by mxaddict** [comment]
Test suite passed

**Comment by mxaddict** [comment]
`coldstaking_fee.py` looks good

**Comment by proletesseract** [comment]
compiles and runs fine. I still need to run the test suite and do a full change/code review.

**Comment by proletesseract** [comment]
coldstaking_fee test passes locally for me on OSX

**Pull request merged**